### PR TITLE
Fix logs not showing up on all servers

### PR DESCRIPTION
**Pull request by wcalderipe**
For some mysterious reason, a traversed object using `.map` is not formatted by Winston, resulting in the logs being swallowed. For further context, we moved to `.map` because it's supposed to clone the node before changing it to fix a mutation issue on the API responses (see #469).

This PR goes back to `.forEach` but ensures nodes that are classes extending from Error are cloned. It also cleans up the noise in the test file by reducing the input samples to a bare minimum.

### CODE REVIEW

File: packages/nestjs-shared/src/lib/module/logger/logger.util.ts [hunk 1]
```diff
@@ -15,7 +15,13 @@ const isSensitiveKey = (key?: string): boolean => {
 export const redact = <T>(input: T): T => {
   const copy = cloneDeep(input)
 
-  return traverse(copy).map(function redactor() {
+  return traverse(copy).forEach(function redactor() {
+    // Prevents a mutation in the original input when the value is a class
+    // extending Error.
+    if (this.node instanceof Error) {
```
Comment (Ptroger) [inline packages/nestjs-shared/src/lib/module/logger/logger.util.ts:21]: Not spending one more minute on this eh ?
I've added this to a list of things I'd like to clarify, this is a very specific behavior and its bugging me.

**Review by Ptroger — APPROVED** [review]
Good job on fixing the thing efficiently

File: packages/nestjs-shared/src/lib/module/logger/logger.util.ts (hunk 1 above)
Comment (wcalderipe) [reply packages/nestjs-shared/src/lib/module/logger/logger.util.ts:21]: To be honest, `.map` works fine until the traversed object gets into winston format [here](https://github.com/narval-xyz/armory/blob/21f2f490167762b2a6ca90a960209b7d43505c0a/packages/nestjs-shared/src/lib/module/logger/winston.logger.ts#L32-L32). So, our initial approach was good but it failed on the integration.

I tried to get to the weeds of it. However, in the format function I can see the value is still an error class not an cloned object.

If I comment out the fix.

```typescript
export const redact = <T>(input: T): T => {
  const copy = cloneDeep(input)

  return traverse(copy).forEach(function redactor() {
    // // Prevents a mutation in the original input when the value is a class
    // // extending Error.
    // if (this.node instanceof Error) {
    //   this.update(clone(this.node))
    // }

    if (isSensitiveKey(this.key)) {
      this.update(REDACT_REPLACE)
    }
  })
}
```

Change the redact format to console log some information

```typescript
const BASE_FORMAT = [
  winston.format.timestamp(),
  winston.format.json(),
  winston.format((info) => {
    const value = redact(info)
    console.log({ value, instanceof: value.obj.foo instanceof Error })
    return redact(info)
  })()
]
```

I run the tests

```bash
npx jest --watch --config packages/nestjs-shared/jest.unit.ts packages/nestjs-shared --verbose 
```

This is what I get

```
    {
      value: {
        obj: { foo: [TestError] },
        level: 'info',
        message: 'foo',
        timestamp: '2024-08-01T15:27:47.221Z',
        [Symbol(level)]: 'info',
        [Symbol(splat)]: [ [Object] ],
        [Symbol(message)]: '{"level":"info","message":"foo","obj":{"foo":{"config":{"secret":"DO NOT MUTATE"}}},"timestamp":"2024-08-01T15:27:47.221Z"}'
      },
      instanceof: true
    }
```
